datafusion/src/logical_plan/builder.rs

```diff
@@ -146,10 +148,12 @@ impl LogicalPlanBuilder {
 
     /// Convert a table provider into a builder with a TableScan
     pub fn scan(
-        table_name: &str,
+        table_name: impl Into<String>,
         provider: Arc<dyn TableProvider>,
         projection: Option<Vec<usize>>,
     ) -> Result<Self> {
+        let table_name = table_name.into();
```

Comment:
This change makes sense to me as this function previously was making an explicit copy and no can reuse the copy the caller passed in, if any 👍 

```
            table_name: table_name.to_string(),
```